api/services/SiteMembershipCreator.js

```diff
@@ -57,13 +58,28 @@ const revokeSiteMembership = ({ user, site, userId }) =>
     .then((permissions) => {
       if (!permissions.push) {
         throw {
-          message: 'You do not have write access to this repository',
+          message: siteErrors.WRITE_ACCESS_REQUIRED,
           status: 400,
         };
       }
     })
     .then(() => {
       const userToRemove = site.Users.find(u => u.id === Number(userId));
+
+      if (!userToRemove) {
+        throw {
+          message: siteErrors.NO_ASSOCIATED_USER,
+          status: 404,
+        };
+      }
+
+      if (userToRemove.username.toLowerCase() === site.owner.toLowerCase()) {
+        throw {
+          message: siteErrors.OWNER_REMOVE,
+          status: 400,
+        };
+      }
+
       site.removeUser(userToRemove);
```

Comment:
As we discussed while pairing, we're pretty sure there needs to be a `return` in front of this statement because it is probably also async. That is, the issue turned out to be a race condition of this removal not completing before the site was refetched and returned to the front end.